packages/vm/benches/main.rs

```diff
@@ -235,14 +237,20 @@ pub fn bench_instance_threads(c: &mut Criterion) {
             unsafe { Cache::new(options).unwrap() };
         let cache = Arc::new(cache);
 
-        let checksum = cache.save_wasm(CONTRACT).unwrap();
+        let checksums: Vec<_> = CONTRACTS
+            .iter()
+            .map(|&wasm| cache.save_wasm(wasm).unwrap())
+            .collect();
+
+        let random_checksum = || *checksums.choose(&mut rand::thread_rng()).unwrap();
 
         b.iter_custom(|iters| {
             let mut res = Duration::from_secs(0);
             for _ in 0..iters {
                 let mut durations: Vec<_> = (0..INSTANTIATION_THREADS)
                     .map(|_id| {
                         let cache = Arc::clone(&cache);
+                        let checksum = random_checksum();
```

Comment:
This can only have two values, right?

Some time ago I created a contract that returns a constant numeric value in a query, let's say 0xF00BAA. Then I used search&replace to find the value in Wasm and other values in there. See https://github.com/cosmos/cosmjs/commit/488dd79be7ad7b2f23e4b23293daa21b327cb7a2

This would allow us to use a very flexible number of checksums starting from a single base contract.

Reply:
Nice. Will add that.